.github/workflows/linux.yml

```diff
@@ -7,6 +7,26 @@ on:
 
 jobs:
 
+   pre-commit:
+      runs-on: ubuntu-latest
+      steps:
+      - uses: actions/checkout@v2
+        with:
+           submodules: recursive
+      - uses: actions/setup-python@v2
+        with:
+           python-version: "3.6"
+      - uses: actions/setup-python@v2
+        with:
+           python-version: "3.7"
+      - uses: actions/setup-python@v2
+        with:
+           python-version: "3.8"
+      - uses: actions/setup-python@v2
+        with:
+           python-version: "3.9"
+      - uses: pre-commit/action@v2.0.0
```

Comment:
why are so many versions of python required for this?

Reply:
There are subtle differences between versions. This is a way to definitively test against all supported python versions. Examples of open source python projects doing this: [youtube-dl](https://github.com/ytdl-org/youtube-dl/blob/master/.github/workflows/ci.yml), [fastapi](https://github.com/tiangolo/fastapi/blob/master/.github/workflows/test.yml), [black](https://github.com/psf/black/blob/master/.github/workflows/test.yml).

Most of the differences between python versions are differences in the standard library, not language features. So this becomes more important as we write more pure python code, rather than just native-code extensions.

Although now that I look more closely at the way open-source projects do this, it looks like they matrix-config the python version. The way I have the mypy pre-config config set up actually requires all python versions to be installed, even when running locally. I'll change that.

Reply:
the linter itself behaves differently on different versions of python?
Still, it looks like the way you invoke it is to just run the liter once though, with one of the python versions

Reply:
yes. `flake8` and `mypy` generally check the target code against the running environment. You can do some cross-version checking with `mypy`, but it's incomplete (if you want to type-check against a third-party package, you need to install the package in a python environment of the desired version)

The multi-version invocation actually happens in `.pre-commit-config.yml`, where there are multiple entries for `mypy`. Hence you need four python installations even to run pre-commit locally. I will change this.